vyper/signatures/function_signature.py

```diff
@@ -219,8 +219,9 @@ def from_definition(cls,
             else:
                 mem_pos += get_size_of_type(parsed_type) * 32
 
-        # Apply decorators
-        const, payable, private, public, nonreentrant_key = False, False, False, False, ''
+        const, payable, private, public, nonreentrant_key = constant, False, False, False, ''
```

Comment:
Maybe there's some subtlety I'm missing but why are there two similarly named variables?

Reply:
constant is a parameter that is is applied to FunctionSignature for Interfaces `def blah(): constant` / `def bah(): modifying`

Reply:
Maybe we can leave a comment explaining the difference? Will save future maintainers some time :)

Reply:
I vote for just renaming the variable to 'constant_override'

Reply:
@jacqueswww @fubuloubu @charles-cooper Yeah, I thought about this as well but I figured I'd just leave it since the `constant` name is part of the external API of `from_definition` and I didn't want to change it.  But I guess it wouldn't be hard to do that and I like the `constant_override` idea.